src/main/java/frc/robot/util/Vision.java

```diff
@@ -82,9 +82,22 @@ public Vision(ArrayList<Pair<String, Transform3d>> camList) {
     for (int i = 0; i < camList.size(); i++) {
       m_cameras.add(new VisionCamera(camList.get(i).getFirst(), camList.get(i).getSecond()));
     }
+    visionLogging();
   }
 
-
+  public void visionLogging() {
+    ArrayList<Pose2d> loggedPoses = new ArrayList<>();
+     if(Constants.DO_LOGGING)  {
+      for(int i = 0; i < m_cameras.size(); i++) {
+      Pose2d pose = getPose2d(loggedPoses.get(i));
+              LogManager.add("Vision/camera " + i + "/estimated pose2d", () -> new Double[] {
```

Comment:
This will not work. loggedPoses will have a length of 0, so you will get an ArrayIndexOutOfBoundsException when deploying. getPose2d() does not get the pose from a camera. Use m_cameras.get(i).getEstimatedPose(referencePose). The reference pose should be the drivetrain's pose, so I'm not sure how you will put that in this method. Also, it will only log a single value because the local variable pose will never be updated. You have to get the pose in the supplier so it is updated.